contracts/DecodeBlock.sol

```diff
@@ -158,7 +356,22 @@ library DecodeBlock {
         array = item.toAddressArray();
     }
 
-    function _decodeNextValidatorPowers(RLPReader.RLPItem memory item) private pure returns (uint64[] memory array) {
-        array = item.toUint64Array();
+    function _decodeNextValidatorPowers(RLPReader.RLPItem memory item) private pure returns (uint256[] memory array) {
+        array = item.toUintArray();
+        return toOnChainPowers(array);
```

Comment:
Looks like may introduce a rounding issue - toOnChainPowers() may be smaller than the actual powers called from the staking contract.  I checked the code and I understand that the powers should be from the header after converting it to on-chain version, but I feel it is bit confusing here.

In fact, I think that we should never use toOnChainPowers() as the conversion to Tendermint should be uni-direction, i.e., we should always use converted powers (toTendermintPowers) for quorum verification.